# Section 1.2 - Exercise 46

In a wind tunnel experiment, the force on a projectile due to air resistance was measured at different velocities:

| Velocity (100 ft/sec) | 0 | 2 | 4 | 6 | 8 | 10 |
|-----------------------|---|---|---|---|---|----|
| Force (100 lb)        | 0 | 2.90 | 14.8 | 39.6 | 74.3 | 119 |

Find an interpolating polynomial for these data and estimate the force on the projectile when the projectile is traveling at **750 ft/sec**.

Use:

\[
p(t)=a_0+a_1t+a_2t^2+a_3t^3+a_4t^4+a_5t^5
\]

What happens if you try to use a polynomial of degree less than 5?  
(Try a cubic polynomial, for instance.)

## Solution

In [1]:
import sympy as sp
A=sp.Matrix([[1, 0, 0, 0, 0, 0],
             [1, 2, 4, 8, 16, 32],
             [1, 4, 16, 64, 256, 1024],
             [1, 6, 36, 216, 1296, 7776],
             [1, 8, 64, 512, 4096, 32768],
             [1, 10, 100, 1000, 10000, 100000]])
b=sp.Matrix([[0, 2.90, 14.8, 39.6, 74.3, 119]])
ans=sp.linsolve((A,b))
print(ans)

{(0, 1.71249999999998, -1.19479166666665, 0.66145833333333, -0.0700520833333329, 0.00260416666666665)}


In [73]:
import numpy as np  

V= np.array([0, 2, 4, 6, 8, 10])  

F= np.array([0, 2.90, 14.8, 39.6, 74.3, 119])  

P5 = np.polyfit(V, F, 5)  

print(P5)  

t = 7.5  

f_1 = np.polyval(P5, t)  

print(f_1)  

F_1= f_1 * 100  

print(F_1)  


[ 2.60416667e-03 -7.00520833e-02  6.61458333e-01 -1.19479167e+00
  1.71250000e+00  2.08856057e-13]
64.83837890625014
6483.837890625015


The interpolation polynomial is:

$$
P(t)=0+1.7125t-1.1948t^2+0.6615t^3-0.0701t^4+0.0026t^5
$$

P(7.5)$\approx$6484(1b)

In [72]:
import numpy as np  

V= np.array([0, 2, 4, 6, 8, 10])  

F= np.array([0, 2.90, 14.8, 39.6, 74.3, 119])  

P3 = np.polyfit(V, F, 3)  

print(P3)  

f_3 = np.polyval(P3, t)  

print(f_3)  

F_3= f_3 * 100  

print(F_3)  


[-0.00462963  1.40962302 -1.7412037   0.17539683]
64.45453869047624
6445.453869047624


A cubic least-squares fit gives approximately:

$$
P3(t)=0.1754-1.7412t+1.4096t^2-0.0046t^3
$$

P3(7.5)$\approx$6445(1b)

A cubic polynomial cannot interpolate all six points simultaneously.

# Section 1.7 - Exercises 47-50

## Exercises 47 and 48

Use as many columns of \(A\) as possible to construct a matrix \(B\) with the property that the equation

\[
Bx=0
\]

has only the trivial solution.

Solve \(Bx=0\) to verify your work.

---

# Section 1.7 - Exercise 47

Given

\[
A=
\begin{bmatrix}
8 & -3 & 0 & -7 & 2\\
-9 & 4 & 5 & 11 & -7\\
6 & -2 & 2 & -4 & 4\\
5 & -1 & 7 & 0 & 10
\end{bmatrix}
\]

Construct a matrix \(B\) using as many columns of \(A\) as possible such that

\[
Bx=0
\]

has only the trivial solution.

Verify by solving \(Bx=0\).

---

# Section 1.7 - Exercise 48

Given

\[
A=
\begin{bmatrix}
12 & 10 & -6 & -3 & 7 & 10\\
-7 & -6 & 4 & 7 & -9 & 5\\
9 & 9 & -9 & -5 & 5 & -1\\
-4 & -3 & 1 & 6 & -8 & 9\\
8 & 7 & -5 & -9 & 11 & -8
\end{bmatrix}
\]

Construct a matrix \(B\) using as many columns of \(A\) as possible such that

\[
Bx=0
\]

has only the trivial solution.

Verify by solving \(Bx=0\).

---

# Section 1.7 - Exercise 49

With \(A\) and \(B\) as in Exercise 47, select a column vector \(v\) of \(A\) that was **not** used in the construction of \(B\).

Determine whether \(v\) is in the set spanned by the columns of \(B\).

Describe your calculations.

---

# Section 1.7 - Exercise 50

Repeat Exercise 49 with the matrices \(A\) and \(B\) from Exercise 48.

Then give an explanation for what you discover, assuming that \(B\) was constructed as specified.

## Solution

### Exercise 47

In [19]:
import sympy as sp
A_47 = sp.Matrix([
    [8, -3, 0, -7, 2],   
    [-9, 4, 5, 11, -7],   
    [6, -2, 2, -4, 4],    
    [5, -1, 7, 0, 10]   
]) 
rref_47= A_47.rref()
print(rref_47)

(Matrix([
[1, 0, 3, 1, 0],
[0, 1, 8, 5, 0],
[0, 0, 0, 0, 1],
[0, 0, 0, 0, 0]]), (0, 1, 4))


Therefore,

In [25]:
B_47 = A_47[:, [0,1,4]]
print(B_47)

Matrix([[8, -3, 2], [-9, 4, -7], [6, -2, 4], [5, -1, 10]])


In [26]:
nullspace_B_47 = B_47.nullspace()
print(nullspace_B_47)

[]


Therefore,The system has only the trivial solution.

### Exercise 48

In [43]:

A_48 = sp.Matrix([
    [12, 10, -6, -3, 7, 10],  
    [-7, -6, 4, 7, -9, 5],     
    [9, 9, -9, -5, 5, -1],     
    [-4, -3, 1, 6, -8, 9],     
    [8, 7, -5, -9, 11, -8]    
]) 

rref_48= A_48.rref()
print(rref_48)

(Matrix([
[1, 0,  2, 0,  2, 0],
[0, 1, -3, 0, -2, 0],
[0, 0,  0, 1, -1, 0],
[0, 0,  0, 0,  0, 1],
[0, 0,  0, 0,  0, 0]]), (0, 1, 3, 5))


Therefore,

In [30]:
B_48 = A_48[:, [0,1,3,5]]
print(B_48)

Matrix([[12, 10, -3, 10], [-7, -6, 7, 5], [9, 9, -5, -1], [-4, -3, 6, 9], [8, 7, -9, -8]])


In [31]:
nullspace_B_48 = B_48.nullspace()
print(nullspace_B_48)

[]


Therefore,The system has only the trivial solution.

### Exercise 49

Choose:
v=[0,5,2,7]

Solve:BX=V

In [81]:
import sympy as sp
B=sp.Matrix([[8, -3, 2], 
            [-9, 4, -7], 
            [6, -2, 4], 
            [5, -1, 10]])
v=sp.Matrix([0,5,2,7])
ans=sp.linsolve((B,v))
print(ans)

{(3, 8, 0)}


The solution is:
x=[3,8,0]

Therefore:
v=a3=3b1+8b2

Hence:

v is in the set spanned by the columns of B

### Exercise 50

Choose:
v=[-6,4,-9,1,-5]

Solve:BX=V

In [1]:
import sympy as sp
B=sp.Matrix([[12, 10, -3, 10], 
             [-7, -6, 7, 5], 
             [9, 9, -5, -1], 
             [-4, -3, 6, 9], 
             [8, 7, -9, -8]])
v=sp.Matrix([-6,4,-9,1,-5])
ans=sp.linsolve((B,v))
print(ans)

{(2, -3, 0, 0)}


The solution is:X=[2,-3,0,0]

Therefore:v=a3=2b1-3b2

Hence:

v is in the set spanned by the columns of B

# Section 1.8 - Exercise 38

An **affine transformation**

\[
T:\mathbb{R}^n\rightarrow \mathbb{R}^m
\]

has the form

\[
T(x)=Ax+b
\]

where \(A\) is an \(m\times n\) matrix and \(b\in\mathbb{R}^m\).

Show that \(T\) is **not a linear transformation** when

\[
b\neq 0
\]

(affine transformations are important in computer graphics).

## Solution

A linear transformation must satisfy:T(0)=0

Now evaluate the T(0):
T(0)=A*0+b=b

When:
$b\neq0$

Therefore:
$
T(0)=b\neq0
$

Therefore:
T(X)=Ax+b is not linear when $b\neq0$

# Section 1.9 - Exercises 45-48

In Exercises 45–48, let \(T\) be the linear transformation whose standard matrix is given.

- In Exercises 45 and 46, decide if \(T\) is a **one-to-one mapping**.
- In Exercises 47 and 48, decide if \(T\) maps \(\mathbb{R}^5\) **onto** \(\mathbb{R}^5\).

Justify your answers.

---

# Section 1.9 - Exercise 45

\[
A=
\begin{bmatrix}
-5 & 10 & -5 & 4\\
8 & 3 & -4 & 7\\
4 & -9 & 5 & -3\\
-3 & -2 & 5 & 4
\end{bmatrix}
\]

Determine whether \(T:\mathbb{R}^4\rightarrow\mathbb{R}^4\) is one-to-one.

---

# Section 1.9 - Exercise 46

\[
A=
\begin{bmatrix}
7 & 5 & 4 & -9\\
10 & 6 & 16 & -4\\
12 & 8 & 12 & 7\\
-8 & -6 & -2 & 5
\end{bmatrix}
\]

Determine whether \(T:\mathbb{R}^4\rightarrow\mathbb{R}^4\) is one-to-one.

---

# Section 1.9 - Exercise 47

\[
A=
\begin{bmatrix}
4 & -7 & 3 & 7 & 5\\
6 & -8 & 5 & 12 & -8\\
-7 & 10 & -8 & -9 & 14\\
3 & -5 & 4 & 2 & -6\\
-5 & 6 & -6 & -7 & 3
\end{bmatrix}
\]

Determine whether

\[
T:\mathbb{R}^5\rightarrow\mathbb{R}^5
\]

maps \(\mathbb{R}^5\) onto \(\mathbb{R}^5\).

---

# Section 1.9 - Exercise 48

\[
A=
\begin{bmatrix}
9 & 13 & 5 & 6 & -1\\
14 & 15 & -7 & -6 & 4\\
-8 & -9 & 12 & -5 & -9\\
-5 & -6 & -8 & 9 & 8\\
13 & 14 & 15 & 2 & 11
\end{bmatrix}
\]

Determine whether

\[
T:\mathbb{R}^5\rightarrow\mathbb{R}^5
\]

maps \(\mathbb{R}^5\) onto \(\mathbb{R}^5\).

## Solution

In [70]:
import sympy as sp

A45 = sp.Matrix([
    [-5,10,-5,4],
    [8,3,-4,7],
    [4,-9,5,-3],
    [-3,-2,5,4]
])

A46 = sp.Matrix([
    [7,5,4,-9],
    [10,6,16,-4],
    [12,8,12,7],
    [-8,-6,-2,5]
])

A47 = sp.Matrix([
    [4,-7,3,7,5],
    [6,-8,5,12,-8],
    [-7,10,-8,-9,14],
    [3,-5,4,2,-6],
    [-5,6,-6,-7,3]
])

A48 = sp.Matrix([
    [9,13,5,6,-1],
    [14,15,-7,-6,4],
    [-8,-9,12,-5,-9],
    [-5,-6,-8,9,8],
    [13,14,15,2,11]
])

A45_rref=A45.rref()[0]
A46_rref=A46.rref()[0]
A47_rref=A47.rref()[0]
A48_rref=A48.rref()[0]

print(A45_rref)
print(A46_rref)
print(A47_rref)
print(A48_rref)

Matrix([[1, 0, 0, 44/35], [0, 1, 0, 79/35], [0, 0, 1, 86/35], [0, 0, 0, 0]])
Matrix([[1, 0, 7, 0], [0, 1, -9, 0], [0, 0, 0, 1], [0, 0, 0, 0]])
Matrix([[1, 0, 0, 5, 0], [0, 1, 0, 1, 0], [0, 0, 1, -2, 0], [0, 0, 0, 0, 1], [0, 0, 0, 0, 0]])
Matrix([[1, 0, 0, 0, 5], [0, 1, 0, 0, -4], [0, 0, 1, 0, 0], [0, 0, 0, 1, 1], [0, 0, 0, 0, 0]])


### Exercise 45

For A45_rref:
There are 4 columns, but only 3 pivots.

Therefore:A45 is not one-to-one

### Exercise 46

For A46_rref:
There are 4 columns, but only 3 pivots.

Therefore:A46 is not one-to-one

### Exercise 47

For A47_rref:
There are 5 rows, but only 4 pivots.

Therefore:A47 is not onto $R^5$

### Exercise 48

For A48_rref:
There are 5 rows, but only 4 pivots.

Therefore:A48 is not onto $R^5$

# Section 1.10 - Exercise 12

**Budget® Rent a Car in Wichita, Kansas** has a fleet of about 500 cars at three locations.

A car rented at one location may be returned to any of the three locations. The fractions of cars returned to the three locations are shown in the matrix below.

The columns represent the location where cars are rented from, and the rows represent the location where cars are returned to.

\[
A=
\begin{bmatrix}
0.97 & 0.05 & 0.10\\
0.00 & 0.90 & 0.05\\
0.03 & 0.05 & 0.85
\end{bmatrix}
\]

where:

- Row 1: Airport returns
- Row 2: East side office returns
- Row 3: West side office returns

and

- Column 1: Cars rented from Airport
- Column 2: Cars rented from East office
- Column 3: Cars rented from West office

---

Suppose that on Monday there are:

\[
x_0=
\begin{bmatrix}
295\\
55\\
150
\end{bmatrix}
\]

cars at:

- Airport: 295 cars
- East side office: 55 cars
- West side office: 150 cars

Find the approximate distribution of cars on Wednesday.

Use the transition matrix:

\[
x_{n+1}=Ax_n
\]

Since Wednesday is two days after Monday:

\[
x_2=A^2x_0
\]

Calculate:

\[
x_1=Ax_0
\]

then

\[
x_2=Ax_1
\]

The resulting vector gives the approximate number of cars at each location on Wednesday.

## Solution

In [85]:
import sympy as sp  

M = sp.Matrix([[0.97, 0.05, 0.10], 
              [0.00, 0.90, 0.05],  
              [0.03, 0.05, 0.85]])

x1 = sp.Matrix([295, 55, 150])  

x2 = M*x1  

x3 = M*x2  

print(x2) 

print(x3)  

Matrix([[303.900000000000], [57.0000000000000], [139.100000000000]])
Matrix([[311.543000000000], [58.2550000000000], [130.202000000000]])


Therefore, Wednesday's approximate distribution is:


Airport:312 cars


East:58 cars


West:130 cars

# Section 1.10 - Exercise 13

Let \(M\) and \(x_0\) be as in Example 3.

## (a)

Compute the population vectors

\[
x_k,\qquad k=1,\ldots,20
\]

using the population model:

\[
x_{k+1}=Mx_k
\]

Discuss what you find.

The sequence of population vectors is obtained by repeatedly multiplying by the transition matrix:

\[
x_1=Mx_0
\]

\[
x_2=Mx_1=M^2x_0
\]

and in general:

\[
x_k=M^kx_0
\]

Compute the vectors for:

\[
k=1,2,\ldots,20
\]

and analyze the long-term population behavior.

---

## (b)

Repeat part (a) using the initial population vector:

\[
x_0=
\begin{bmatrix}
350000\\
650000
\end{bmatrix}
\]

where:

- City population: \(350,000\)
- Suburban population: \(650,000\)

Compute:

\[
x_k=M^kx_0,\qquad k=1,\ldots,20
\]

Compare the results with part (a) and discuss what happens to the population distribution over time.

## Solution

### (a)

In [86]:
import sympy as sp

M=sp.Matrix([[0.95,0.03],
           [0.05,0.97]])

x=sp.Matrix([600000,400000])

for i in range(1,21):
    x= M * x
    print(i,x)
    

1 Matrix([[582000.000000000], [418000.000000000]])
2 Matrix([[565440.000000000], [434560.000000000]])
3 Matrix([[550204.800000000], [449795.200000000]])
4 Matrix([[536188.416000000], [463811.584000000]])
5 Matrix([[523293.342720000], [476706.657280000]])
6 Matrix([[511429.875302400], [488570.124697600]])
7 Matrix([[500515.485278208], [499484.514721792]])
8 Matrix([[490474.246455951], [509525.753544049]])
9 Matrix([[481236.306739475], [518763.693260525]])
10 Matrix([[472737.402200317], [527262.597799683]])
11 Matrix([[464918.410024292], [535081.589975708]])
12 Matrix([[457724.937222348], [542275.062777651]])
13 Matrix([[451106.942244561], [548893.057755439]])
14 Matrix([[445018.386864996], [554981.613135004]])
15 Matrix([[439416.915915796], [560583.084084204]])
16 Matrix([[434263.562642532], [565736.437357467]])
17 Matrix([[429522.477631130], [570477.522368870]])
18 Matrix([[425160.679420639], [574839.320579360]])
19 Matrix([[421147.825066988], [578852.174933012]])
20 Matrix([[417455.99

The results show:  
City population decreases over time.  
Suburban population increases over time.  
The system approaches a stable distribution.  

### (b)

In [88]:
import sympy as sp 

M = sp.Matrix([[0.95, 0.03],  
              [0.05, 0.97]])

x = sp.Matrix([350000, 650000])  

for i in range(1, 21):  
    x = M * x  
    
    print(i,x)  

1 Matrix([[352000.000000000], [648000.000000000]])
2 Matrix([[353840.000000000], [646160.000000000]])
3 Matrix([[355532.800000000], [644467.200000000]])
4 Matrix([[357090.176000000], [642909.824000000]])
5 Matrix([[358522.961920000], [641477.038080000]])
6 Matrix([[359841.124966400], [640158.875033600]])
7 Matrix([[361053.834969088], [638946.165030912]])
8 Matrix([[362169.528171561], [637830.471828439]])
9 Matrix([[363195.965917836], [636804.034082164]])
10 Matrix([[364140.288644409], [635859.711355590]])
11 Matrix([[365009.065552856], [634990.934447143]])
12 Matrix([[365808.340308628], [634191.659691372]])
13 Matrix([[366543.673083937], [633456.326916062]])
14 Matrix([[367220.179237222], [632779.820762777]])
15 Matrix([[367842.564898245], [632157.435101755]])
16 Matrix([[368415.159706385], [631584.840293614]])
17 Matrix([[368941.946929874], [631058.053070125]])
18 Matrix([[369426.591175484], [630573.408824515]])
19 Matrix([[369872.463881445], [630127.536118554]])
20 Matrix([[370282.66

The system also approaches a stable distribution. 

Now caculate the stable state:  

In [92]:
import sympy as sp 
M = sp.Matrix([[0.95, 0.03],  
              [0.05, 0.97]])

x = sp.Matrix([600000, 400000])  

for i in range(1, 1000):  
    x = M * x  
    
print(x)  

Matrix([[374999.999999994], [624999.999999991]])


In [93]:
import sympy as sp 
M = sp.Matrix([[0.95, 0.03],  
              [0.05, 0.97]])

x = sp.Matrix([350000, 650000])  

for i in range(1, 1000):  
    x = M * x  
    
print(x)  

Matrix([[374999.999999994], [624999.999999993]])


Therefore:  
The long-term distribution is:  
37.5% city and 62.5% suburbs

# Section 1.3 - Exercise 35

A mining company has two mines.

One day's operation at mine 1 produces ore that contains:

- 20 metric tons of copper
- 550 kilograms of silver

One day's operation at mine 2 produces ore that contains:

- 30 metric tons of copper
- 500 kilograms of silver


Let:

$$
v_1=
\begin{bmatrix}
20\\
550
\end{bmatrix}
$$


and


$$
v_2=
\begin{bmatrix}
30\\
500
\end{bmatrix}
$$


Then \(v_1\) and \(v_2\) represent the output per day of mine 1 and mine 2.

---

## (a)

What physical interpretation can be given to:

$$
5v_1
$$


---

## (b)

Suppose the company operates mine 1 for \(x_1\) days and mine 2 for \(x_2\) days.

Write a vector equation whose solution gives the number of days each mine should operate in order to produce:

- 150 metric tons of copper
- 2825 kilograms of silver

Do not solve the equation.

---

## (c)

Solve the equation in part (b).

## Solution

### (a)

We have:

In [2]:
import sympy as sp
v1=sp.Matrix([20,550])
print(5*v1)

Matrix([[100], [2750]])


The vector 5v1 represents the output of mine 1 operating for 5 days.  
During these 5 days, mine 1 produces:  
100 metric tons of copper  
2750 kilograms of silver  

### (b) and (c)

In [3]:

import sympy as sp

A = sp.Matrix([
    [20, 30],
    [550, 500]
])

b = sp.Matrix([
    150,
    2825
])

Ab = A.row_join(b)
rref_matrix, pivot_columns = Ab.rref()
print(rref_matrix)

Matrix([[1, 0, 3/2], [0, 1, 4]])


Therefore:  
- x1=1.5  
- x2=4

Mine 1 operates for 1.5 days,and mine 2 operatres for 4 days.

# Section 1.3 - Exercise 36

A steam plant burns two types of coal:

- anthracite (A)
- bituminous (B)


For each ton of A burned, the plant produces:

- 27.6 million Btu of heat
- 3100 grams of sulfur dioxide
- 250 grams of particulate matter


For each ton of B burned, the plant produces:

- 30.2 million Btu of heat
- 6400 grams of sulfur dioxide
- 360 grams of particulate matter


Let \(x_1\) be tons of A and \(x_2\) be tons of B.

---

## (a)

How much heat does the steam plant produce?


---

## (b)

Express the output as a linear combination of two vectors.


---

## (c)

The plant produced:

- 162 million Btu of heat
- 23610 g sulfur dioxide
- 1623 g particulate matter

Determine how many tons of each type of coal were burned.

## Solution

### (a)

Anthracite produces:27.6x1 million Btu

Bituminous produces:30.2x2 million Btu

Therefore: H=27.6x1+30.2x2 million Btu

### (b)

In [4]:
import sympy as sp
# For anthracite:
A=sp.Matrix([27.6,3100,250])
# For bituminous:
B=sp.Matrix([30.2,6400,360])

The total output is:  
x1*A+x2*B

### (c)

In [5]:

import sympy as sp

A = sp.Matrix([
    [27.6, 30.2, 162],
    [3100, 6400, 23610],
    [250, 360, 1623]
])

rref_matrix, pivot_columns = A.rref()
print(rref_matrix)

Matrix([[1, 0, 39/10], [0, 1, 9/5], [0, 0, 0]])


Therefore:  
Anthracite burned: 3.9 tons  
Bituminous coal burned: 1.8 tons  

# Section 1.4 - Exercises 47-50

Determine if the columns of the matrix span $ R^4 $.

---

# Section 1.4 - Exercise 47

$$
A=
\begin{bmatrix}
7&2&-5&8\\
-5&-3&4&-9\\
6&10&-2&7\\
-7&9&2&15
\end{bmatrix}
$$


---

# Section 1.4 - Exercise 48

$$
A=
\begin{bmatrix}
5&-7&-4&9\\
6&-8&-7&5\\
4&-4&-9&-9\\
-9&11&16&7
\end{bmatrix}
$$


---

# Section 1.4 - Exercise 49

$$
A=
\begin{bmatrix}
12&-7&11&-9&5\\
-9&4&-8&7&-3\\
-6&11&-7&3&-9\\
4&-6&10&-5&12
\end{bmatrix}
$$


---

# Section 1.4 - Exercise 50

$$
A=
\begin{bmatrix}
8&11&-6&-7&13\\
-7&-8&5&6&-9\\
11&7&-7&-9&-6\\
-3&4&1&8&7
\end{bmatrix}
$$

## Solution

In [6]:

import sympy as sp

A47 = sp.Matrix([
    [7, 2, -5, 8],
    [-5, -3, 4, -9],
    [6, 10, -2, 7],
    [-7, 9, 2, 15]
])

A48 = sp.Matrix([
    [5, -7, -4, 9],
    [6, -8, -7, 5],
    [4, -4, -9, -9],
    [-9, 11, 16, 7]
])

A49 = sp.Matrix([
    [12, -7, 11, -9, 5],
    [-9, 4, -8, 7, -3],
    [-6, 11, -7, 3, -9],
    [4, -6, 10, -5, 12]
])


A50 = sp.Matrix([
    [8, 11, -6, -7, 13],
    [-7, -8, 5, 6, -9],
    [11, 7, -7, -9, -6],
    [-3, 4, 1, 8, 7]
])

A47_rref_matrix, A47_pivot_columns = A47.rref()
A48_rref_matrix, A48_pivot_columns = A48.rref()
A49_rref_matrix, A49_pivot_columns = A49.rref()
A50_rref_matrix, A50_pivot_columns = A50.rref()

print(A47_rref_matrix,A47_pivot_columns)
print(A48_rref_matrix,A48_pivot_columns)
print(A49_rref_matrix,A49_pivot_columns)
print(A50_rref_matrix,A50_pivot_columns)

Matrix([[1, 0, 0, -93/50], [0, 1, 0, 53/50], [0, 0, 1, -189/50], [0, 0, 0, 0]]) (0, 1, 2)
Matrix([[1, 0, 0, 4/3], [0, 1, 0, -5/3], [0, 0, 1, 7/3], [0, 0, 0, 0]]) (0, 1, 2)
Matrix([[1, 0, 0, -10/21, 0], [0, 1, 0, -25/84, 0], [0, 0, 1, -41/84, 0], [0, 0, 0, 0, 1]]) (0, 1, 2, 4)
Matrix([[1, 0, -7/13, 0, 0], [0, 1, -2/13, 0, 0], [0, 0, 0, 1, 0], [0, 0, 0, 0, 1]]) (0, 1, 3, 4)


### Exercise 47

Row reduction gives only 3 pivot positions.  
There is no pivot in every row, so the columns do not span $ R^4 $.

### Exercise 48

Row reduction gives only 3 pivot positions.  
There is no pivot in every row, so the columns do not span $ R^4 $.

### Exercise 49

Row reduction gives 4 pivot positions.  
There is a pivot in every row, so the columns span $ R^4$.

### Exercise 50

Row reduction gives 4 pivot positions.  
There is a pivot in every row, so the columns span $ R^4$.